src/main/java/de/ait/secondlife/controllers/ImageContorller.java

```diff
@@ -0,0 +1,90 @@
+import de.ait.secondlife.domain.dto.ResponseMessageDto;
+import de.ait.secondlife.exception_handling.exceptions.not_found_exception.BadEntityTypeException;
+import de.ait.secondlife.services.interfaces.ImageService;
+import de.ait.secondlife.services.utilities.EntityUtilities;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.media.Content;
+import io.swagger.v3.oas.annotations.media.Schema;
+import io.swagger.v3.oas.annotations.responses.ApiResponse;
+import io.swagger.v3.oas.annotations.responses.ApiResponses;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+@RestController
+@RequestMapping("/v1/images")
+@RequiredArgsConstructor
+@Tag(name = "Image controller", description = "Controller for some operations with images")
+public class ImageContorller {
+
+    private final ImageService imageService;
+    private final EntityUtilities utilities;
+
+    @PostMapping
+    @Operation(
+            summary = "Create new image",
+            description = "Create new image for entity by entity id. " +
+                    "The maximum allowable image upload size is 8MB. " +
+                    "The maximum number of images for the Offer entity is 5. " +
+                    "For Category and User - 1."
+    )
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Successful operation",
+                    content = @Content(mediaType = "application/json",
+                            schema = @Schema(implementation = ImagePathsResponseDto.class))),
+            @ApiResponse(responseCode = "400", description = "Bad request", content = @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = ResponseMessageDto.class)
+            )),
+            @ApiResponse(responseCode = "404", description = "Resource not found", content = @Content(
+                    mediaType = "application/json",
+                    schema = @Schema(implementation = ResponseMessageDto.class)
+            ))})
+    public ResponseEntity<ImagePathsResponseDto> uploadImage(
+                                                           @Valid
+                                                           @Parameter(description = "Dto with image file, entity type and entity id ", schema = @Schema(implementation = ImageCreationDto.class))
+                                                           ImageCreationDto request) {
+
+        String entityType = EntityTypeWithImgs.get(request.getEntityType().toLowerCase()).getType();
+        Long entityId = request.getEntityId();
+        if (!utilities.checkEntityById(entityType, entityId))
+            throw new BadEntityTypeException(entityType, entityId);
```

Comment:
could we move this business logic from controller to service or utility class?
and "boolean checkSomething()" is not an obvious method name. what about "isEntityExists()" ?

Reply:
If the call to this method is in a service, it causes a cyclic dependency. I had to put this check in the controller. The logic of the check has been removed to the utility The method has been renamed